### PR TITLE
Add kwargs to _moments to support additional Keras parameters

### PR DESCRIPTION
**Pull request by Tixxx**
Signed-off-by: TJ <tjx@nvidia.com>

## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description
Add kwargs to support addition parameters added by keras.layers.batchnormalization.

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by EnricoMi** [comment]
We are now seeing

    [0]<stderr>:Node: 'sequential/conv2d/Relu'
    [0]<stderr>:DNN library is not found.
    [0]<stderr>:     [[{{node sequential/conv2d/Relu}}]] [Op:__inference_training_step_1359]

https://buildkite.com/horovod/horovod/builds/8663#018489d6-74e6-4fb7-b91c-b4952ab5e9a0/232-367

Shall we fix this in a separate PR?

**Comment by Tixxx** [comment]
> We are now seeing
> 
> ```
> [0]<stderr>:Node: 'sequential/conv2d/Relu'
> [0]<stderr>:DNN library is not found.
> [0]<stderr>:     [[{{node sequential/conv2d/Relu}}]] [Op:__inference_training_step_1359]
> ```
> 
> https://buildkite.com/horovod/horovod/builds/8663#018489d6-74e6-4fb7-b91c-b4952ab5e9a0/232-367
> 
> Shall we fix this in a separate PR?

found this line in the logs:
`Loaded runtime CuDNN library: 8.4.1 but source was compiled with: 8.6.0.`
Let me try changing the cudnn version to 8.6, if it doesnt fix it then i will open another pr to address it.

**Comment by EnricoMi** [comment]
That would be [libcudnn8-dev_8.6.0.163-1+cuda11.8_amd64.deb](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/libcudnn8-dev_8.6.0.163-1+cuda11.8_amd64.deb)
https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/

**Comment by Tixxx** [comment]
> That would be [libcudnn8-dev_8.6.0.163-1+cuda11.8_amd64.deb](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/libcudnn8-dev_8.6.0.163-1+cuda11.8_amd64.deb) https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/

Only cuda11.8 is available with cudnn 8.6. Should I also change the base to `11.8.0-devel-ubuntu20.04` ?

**Review by chongxiaoc — APPROVED** [review]

File: docker-compose.test.yml [hunk 1]
```diff
@@ -225,13 +225,13 @@ services:
     extends: test-gpu-base
     build:
       args:
-        CUDA_DOCKER_VERSION: 11.6.1-devel-ubuntu20.04
-        CUDNN_VERSION: 8.4.1.50-1+cuda11.6
+        CUDA_DOCKER_VERSION: 11.8.0-devel-ubuntu20.04
+        CUDNN_VERSION: 8.6.0.163-1+cuda11.8
         NCCL_VERSION_OVERRIDE: 2.11.4-1+cuda11.6
```
Comment (EnricoMi) [inline docker-compose.test.yml:230]: should we also move NCCL to cuda 11.8, e.g. [libnccl2_2.15.5-1+cuda11.8_amd64.deb](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/libnccl2_2.15.5-1+cuda11.8_amd64.deb)?

**Review by EnricoMi — APPROVED** [review]
Minor comment, LGTM!

File: docker-compose.test.yml (hunk 1 above)
Comment (Tixxx) [reply docker-compose.test.yml:230]: yep, i think we should.

Comment (Tixxx) [reply docker-compose.test.yml]: hmm looks like the new nccl has some issues, seeing this in the tests: 
`"[1]<stdout>:E     tensorflow.python.framework.errors_impl.UnknownError: {{function_node __wrapped__HorovodAllgather_device_/job:localhost/replica:0/task:0/device:GPU:0}} ncclAllGather failed: invalid argument [Op:HorovodAllgather] name: test_start.sz"`
will have to upgrade it in another separate pr then.

**Comment by EnricoMi** [comment]
NCCL upgrade was still worth trying. Thanks for fixing this, this is ready to be merged then.

**Comment by Tixxx** [comment]
> NCCL upgrade was still worth trying. Thanks for fixing this, this is ready to be merged then.

Agreed. I'm tracking it in a separate issue [here](https://github.com/horovod/horovod/issues/3779).

**Pull request merged**